src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs

```diff
@@ -116,19 +186,77 @@ public void Dump()
 
             public void Clear()
             {
-                _objectsCache.Clear();
+                _byHashObjectCache.Clear();
+                _byKeyObjectCache.Clear();
                 Interlocked.Exchange(ref _count, 0);
                 Metrics.CachedNodesCount = 0;
                 _trieStore.MemoryUsedByDirtyCache = 0;
             }
+
+            internal readonly struct Key : IEquatable<Key>
+            {
+                internal const long MemoryUsage = 8 + 36 + 8; // (address (probably shared), path, keccak pointer (shared with TrieNode))
+                public Hash256? Address { get; }
+                public TreePath Path { get; }
+                public Hash256 Keccak { get; }
+
+                public Key(Hash256? address, in TreePath path, Hash256 keccak)
+                {
+                    Address = address;
+                    Path = path;
+                    Keccak = keccak;
+                }
+
+                public override int GetHashCode()
+                {
+                    return HashCode.Combine(Keccak.GetHashCode(), Address?.GetHashCode() ?? 0, Path.GetHashCode());
```

Comment:
Isn't it something that pops up during profiling? `HashCode.Combine` can be quite heavy sometimes.

Reply:
Hmm... fair point. I originally, just use the Keccak. But it turns out a small number of `Keccak` is used on multiple path, see #6752. Hard to know. Maybe just `Path` and `Keccak` is enough?

Reply:
Copied code from paprika.